module/mempool/herocache/backdata/cache_test.go

```diff
@@ -125,6 +126,113 @@ func TestArrayBackData_Adjust(t *testing.T) {
 	require.Equal(t, bd.Size(), uint(limit))
 }
 
+// TestRemoveAfterAdjustRandom ensures that when you Adjust a random entry in a full cache,
+// Remove returns the updated entity, the cache size drops by one, and all other entries remain.
+func TestArrayBackData_RemoveAfterAdjustRandom(t *testing.T) {
+	limit := 100_000
+
+	bd := NewCache[*unittest.MockEntity](
+		uint32(limit),
+		8,
+		heropool.LRUEjection,
+		unittest.Logger(),
+		metrics.NewNoopCollector(),
+	)
+
+	entities := unittest.EntityListFixture(uint(limit))
+
+	// adds all entities to backdata
+	testAddEntities(t, bd, entities, heropool.LRUEjection)
+
+	// pick one at random and Adjust it
+	entityIndex := rand.Int() % limit
+	original, ok := bd.Get(entities[entityIndex].Identifier)
+	require.True(t, ok)
+
+	// adjusts old entity to a new entity
+	updatedEntity, ok := bd.Adjust(original.Identifier, func(ent *unittest.MockEntity) *unittest.MockEntity {
+		require.True(t, ok)
+
```

Comment:
```suggestion
```
Since `Adjust` has not yet returned here I think we're just re-checking the assignment from line 149?